openhands/runtime/impl/remote/remote_runtime.py

```diff
@@ -318,7 +318,7 @@ def _wait_until_alive(self):
             raise RuntimeError(msg)
 
     def close(self, timeout: int = 10):
-        if self.config.sandbox.keep_remote_runtime_alive:
+        if self.config.sandbox.keep_remote_runtime_alive or self.attach_to_existing:
```

Comment:
we maybe need something in `impl/modal` too?

Reply:
I added this to EventStream (Line 409) and Remote (Above) - Modal already has this functionality! https://github.com/All-Hands-AI/OpenHands/blob/8d2b2d431832cb58abad0cd807dfe5742fff1ebb/openhands/runtime/impl/modal/modal_runtime.py#L284